app/products/calls/screens/call_screen/call_screen.tsx

```diff
@@ -0,0 +1,545 @@
+            if (data.channelId === currentCall?.channelId && ((speaker === data.userId) || !speaker)) {
+                setSpeaker('');
+            }
+        };
+
+        const onVoiceOn = DeviceEventEmitter.addListener(WebsocketEvents.CALLS_USER_VOICE_ON, handleVoiceOn);
+        const onVoiceOff = DeviceEventEmitter.addListener(WebsocketEvents.CALLS_USER_VOICE_OFF, handleVoiceOff);
+        return () => {
+            onVoiceOn.remove();
+            onVoiceOff.remove();
+        };
+    }, []);
+
+    const leaveCallHandler = useCallback(() => {
+        popTopScreen();
+        leaveCall();
+    }, []);
+
+    const muteUnmuteHandler = useCallback(() => {
+        if (myParticipant?.muted) {
+            unmuteMyself();
+        } else {
+            muteMyself();
+        }
+    }, [myParticipant?.muted]);
+
+    const toggleRaiseHand = useCallback(() => {
+        const raisedHand = myParticipant?.raisedHand || 0;
+        if (raisedHand > 0) {
+            unraiseHand();
+        } else {
+            raiseHand();
+        }
+    }, [myParticipant?.raisedHand]);
+
+    const toggleControlsInLandscape = useCallback(() => {
+        setShowControlsInLandscape(!showControlsInLandscape);
+    }, [showControlsInLandscape]);
+
+    const switchToThread = useCallback(async () => {
+        Keyboard.dismiss();
+        await dismissBottomSheet();
+        if (!currentCall) {
+            return;
+        }
+
+        const activeUrl = await DatabaseManager.getActiveServerUrl();
+        if (activeUrl === currentCall.serverUrl) {
+            goToScreen(Screens.THREAD, '', {rootId: currentCall.threadId});
+            return;
+        }
+
+        // TODO: this is a temporary solution until we have a proper cross-team thread view.
+        //  https://mattermost.atlassian.net/browse/MM-45752
+        popTopScreen();
+        await DatabaseManager.setActiveServerDatabase(currentCall.serverUrl);
+        await appEntry(currentCall.serverUrl, Date.now());
+        goToScreen(Screens.THREAD, '', {rootId: currentCall.threadId});
+    }, [currentCall?.serverUrl, currentCall?.threadId]);
+
+    const showOtherActions = useCallback(() => {
+        const renderContent = () => {
+            return (
+                <View style={style.bottomSheet}>
+                    <SlideUpPanelItem
+                        icon='message-text-outline'
+                        onPress={switchToThread}
+                        text='Chat thread'
+                    />
+                </View>
+            );
+        };
+
+        bottomSheet({
+            closeButtonId: 'close-other-actions',
+            renderContent,
+            snapPoints: [bottomSheetSnapPoint(2, ITEM_HEIGHT, insets.bottom), 10],
+            title: intl.formatMessage({id: 'post.options.title', defaultMessage: 'Options'}),
+            theme,
+        });
+    }, [insets, intl, theme]);
+
+    if (!currentCall || !myParticipant) {
+        return null;
+    }
```

Comment:
worryiiiing, if for any reason either of this two become false, then this screen will not render anything but it will be on top of the stack, preventing all gestures and render the app unusable, in this case you should pop/dismiss this screen

Reply:
Yes good point. I originally thought I wouldn't need this check, but it looks like it depends on a different PR (and then I have to do a bit of refactoring). Added the correct `pop` and tested it, it works. Added a todo and a new ticket to fix this in a better way: 50d02fc2d55999db71916081b71de08be97bdb9a